pkg/registry/common/dial/nse_client.go

```diff
@@ -94,19 +95,34 @@ func (c *dialNSEClient) Unregister(ctx context.Context, in *registry.NetworkServ
 		return next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, in, opts...)
 	}
 
-	cc, _ := clientconn.Load(ctx)
+	cc, loaded := clientconn.LoadOrStore(ctx, newDialer(c.chainCtx, c.dialTimeout, c.dialOptions...))
 
 	di, ok := cc.(*dialer)
 	if !ok {
 		return next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, in, opts...)
 	}
+
+	if loaded {
+		defer func() {
+			_ = di.Close()
+			clientconn.Delete(ctx)
+		}()
+		return next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, in, opts...)
```

Comment:
Shouldn't we also call `di.Dial(...)` in the case of `loaded`?

Reply:
Done